srml/operator/src/lib.rs

```diff
@@ -0,0 +1,141 @@
+            let operator = ensure_signed(origin)?;
+
+            // verify parameters.
+            parameters.verify()?;
+
+            let contract = T::DetermineContractAddress::contract_address_for(&code_hash, &data, &operator);
+            contract::Module::<T>::instantiate(RawOrigin::Signed(operator.clone()).into(), endowment, gas_limit, code_hash, data)?;
+
+            // add operator to contracts
+            <OperatorHasContracts<T>>::mutate(&operator, {|tree| (*tree).insert(contract.clone()) });
+            // add contract to operator
+            <ContractHasOperator<T>>::insert(&contract, operator.clone());
+            // add contract to parameters
+            <ContractParameters<T>>::insert(&contract, parameters);
+
+            // issue an event operator -> contract
+            Self::deposit_event(RawEvent::SetOperator(operator, contract));
+            Ok(())
+        }
+
+        /// Updates parameters for an identified contact.
+        pub fn update_parameters(origin, contract: T::AccountId, parameters: T::Parameters) -> Result {
+            let operator = ensure_signed(origin)?;
+
+            // verify parameters
+            parameters.verify()?;
+
+            let contracts = <OperatorHasContracts<T>>::get(&operator);
+
+            // check the actually operate the contract.
+            if !contracts.contains(&contract) {
+                return Err("The sender don't operate the contract address.")
+            }
+
+            // update parameters
+            <ContractParameters<T>>::insert(&contract, parameters.clone());
+            // issue set parameter events
+            Self::deposit_event(RawEvent::SetParameters(contract, parameters));
+            Ok(())
+        }
+
+        /// Changes an operator for identified contracts.
+        pub fn change_operator(origin, contracts: Vec<T::AccountId>, new_operator: T::AccountId) -> Result {
+            let operator = ensure_signed(origin)?;
+            let operate_contracts = <OperatorHasContracts<T>>::get(&operator);
+
+            // check the actually operate the contract.
+            for c in contracts.iter() {
+                if !operate_contracts.contains(&c) {
+                    return Err("The sender don't operate the contracts address.")
+                }
+            }
+
+            // remove origin operator to contracts
+            <OperatorHasContracts<T>>::mutate(&operator,
+                |tree| for c in contracts.iter() { (*tree).remove(c); }
+            );
+            // add new_operator to contracts
+            <OperatorHasContracts<T>>::mutate(&new_operator,
+                |tree| for c in contracts.iter() { (*tree).insert(c.clone()); }
+            );
+            for c in contracts.iter() {
+				// add contract to new_operator
+				<ContractHasOperator<T>>::insert(&c, new_operator.clone());
+				// issue an event operator -> contract
+                Self::deposit_event(RawEvent::SetOperator(new_operator.clone(), c.clone()));
```

Comment:
Spaces instead of tabulation.

> What style you prefer? Basically Substrate code use TAB instead of 4 spaces, but in my opinion TAB is a bit uncomfortable to use in everyday life.

Reply:
Thanks. I'm missing format.